cmd/crc/cmd/start.go

```diff
@@ -190,20 +193,39 @@ func validateStartFlags() error {
 	return nil
 }
 
-func checkIfNewVersionAvailable(noUpdateCheck bool) {
+func checkIfNewVersionAvailable(noUpdateCheck bool) error {
 	if noUpdateCheck {
-		return
+		return nil
 	}
-	isNewVersionAvailable, newVersion, err := crcversion.NewVersionAvailable()
+	isNewVersionAvailable, newVersion, link, err := newVersionAvailable()
 	if err != nil {
-		logging.Debugf("Unable to find out if a new version is available: %v", err)
-		return
+		return err
 	}
 	if isNewVersionAvailable {
-		logging.Warnf("A new version (%s) has been published on %s", newVersion, constants.CrcLandingPageURL)
-		return
+		logging.Warnf("A new version (%s) has been published on %s", newVersion, link)
+		return nil
 	}
 	logging.Debugf("No new version available. The latest version is %s", newVersion)
+	return nil
+}
+
+func newVersionAvailable() (bool, string, string, error) {
```

Comment:
All looks good but why this function moved from pkg/version to here, I think it's good that it is part of pkg/version and not scoped to `cmd` only?

Reply:
```

func downloadLink(release *crcversion.CrcReleaseInfo) string {
	if link, ok := release.Links[runtime.GOOS]; ok {
		return link
	}
	return constants.CrcLandingPageURL
}
```

requires a constant in constants pkg. 
But also, versions pkg in included in constants pkg. So there is cyclic dependency if I put this code in version pkg.

Reply:
@guillaumerose Ah yes, ideally we shouldn't import anything related to crc pkg so it always immune to cyclic dependency but looking at the code it looks like unavoidable until we refactor around the way we pass `ldflags` during compilation. In the meanwhile I think following is also clean.
```patch
diff --git a/cmd/crc/cmd/start.go b/cmd/crc/cmd/start.go
index 3ae37714..02863baa 100644
--- a/cmd/crc/cmd/start.go
+++ b/cmd/crc/cmd/start.go
@@ -10,7 +10,6 @@ import (
        "strings"
        "text/template"
 
-       "github.com/Masterminds/semver/v3"
        "github.com/code-ready/crc/pkg/crc/cluster"
        crcConfig "github.com/code-ready/crc/pkg/crc/config"
        "github.com/code-ready/crc/pkg/crc/constants"
@@ -198,7 +197,7 @@ func checkIfNewVersionAvailable(noUpdateCheck bool) error {
        if noUpdateCheck {
                return nil
        }
-       isNewVersionAvailable, newVersion, link, err := newVersionAvailable()
+       isNewVersionAvailable, newVersion, link, err := crcversion.NewVersionAvailable()
        if err != nil {
                return err
        }
@@ -210,25 +209,6 @@ func checkIfNewVersionAvailable(noUpdateCheck bool) error {
        return nil
 }
 
-func newVersionAvailable() (bool, string, string, error) {
-       release, err := crcversion.GetCRCLatestVersionFromMirror()
-       if err != nil {
-               return false, "", "", err
-       }
-       currentVersion, err := semver.NewVersion(crcversion.GetCRCVersion())
-       if err != nil {
-               return false, "", "", err
-       }
-       return release.Version.CrcVersion.GreaterThan(currentVersion), release.Version.CrcVersion.String(), downloadLink(release), nil
-}
-
-func downloadLink(release *crcversion.CrcReleaseInfo) string {
-       if link, ok := release.Links[runtime.GOOS]; ok {
-               return link
-       }
-       return constants.CrcLandingPageURL
-}
-
 const startTemplate = `Started the OpenShift cluster.
 
 The server is accessible via web console at:
diff --git a/pkg/crc/constants/constants.go b/pkg/crc/constants/constants.go
index 98a9d62b..3ba00977 100644
--- a/pkg/crc/constants/constants.go
+++ b/pkg/crc/constants/constants.go
@@ -25,7 +25,6 @@ const (
        ConfigFile                = "crc.json"
        LogFile                   = "crc.log"
        DaemonLogFile             = "crcd.log"
-       CrcLandingPageURL         = "https://cloud.redhat.com/openshift/create/local" // #nosec G101
        DefaultPodmanURLBase      = "https://storage.googleapis.com/libpod-master-releases"
        DefaultAdminHelperCliBase = "https://github.com/code-ready/admin-helper/releases/download/0.0.4"
        CRCMacTrayDownloadURL     = "https://github.com/code-ready/tray-macos/releases/download/v%s/crc-tray-macos.tar.gz"
@@ -42,6 +41,8 @@ const (
        AppsDomain    = ".apps-crc.testing"
 )
 
+var CrcLandingPageURL = version.CrcLandingPageURL
+
 var adminHelperExecutableForOs = map[string]string{
        "darwin":  "crc-admin-helper-darwin",
        "linux":   "crc-admin-helper-linux",
diff --git a/pkg/crc/version/version.go b/pkg/crc/version/version.go
index d73b4991..cab0e293 100644
--- a/pkg/crc/version/version.go
+++ b/pkg/crc/version/version.go
@@ -5,6 +5,7 @@ import (
        "fmt"
        "io/ioutil"
        "net/http"
+       "runtime"
        "time"
 
        "github.com/Masterminds/semver/v3"
@@ -34,6 +35,7 @@ const (
        crcMacTrayVersion = "1.0.9"
        // Windows forms application version type major.minor.buildnumber.revesion
        crcWindowsTrayVersion = "0.6.0.0"
+       CrcLandingPageURL     = "https://cloud.redhat.com/openshift/create/local" // #nosec G101
 )
 
 type CrcReleaseInfo struct {
@@ -109,3 +111,22 @@ func GetCRCLatestVersionFromMirror() (*CrcReleaseInfo, error) {
 
        return &releaseInfo, nil
 }
+
+func NewVersionAvailable() (bool, string, string, error) {
+       release, err := GetCRCLatestVersionFromMirror()
+       if err != nil {
+               return false, "", "", err
+       }
+       currentVersion, err := semver.NewVersion(GetCRCVersion())
+       if err != nil {
+               return false, "", "", err
+       }
+       return release.Version.CrcVersion.GreaterThan(currentVersion), release.Version.CrcVersion.String(), downloadLink(release), nil
+}
+
+func downloadLink(release *CrcReleaseInfo) string {
+       if link, ok := release.Links[runtime.GOOS]; ok {
+               return link
+       }
+       return CrcLandingPageURL
+}

```

Reply:
```
+var CrcLandingPageURL = version.CrcLandingPageURL
```

I dont really like this. It is complicated to follow. Maybe we should merge version and constants pkg ?